crates/forge/tests/e2e/trace_resources.rs

```diff
@@ -0,0 +1,325 @@
+use assert_fs::TempDir;
+use std::fs;
+
+use forge_runner::trace_data::ProfilerDeprecatedSyscallSelector::{
+    CallContract, Deploy, EmitEvent, GetBlockHash, GetExecutionInfo, Keccak, LibraryCall,
+    SendMessageToL1, StorageRead, StorageWrite,
+};
+use forge_runner::trace_data::{ProfilerCallTrace, ProfilerExecutionResources, TRACE_DIR};
+
+use crate::e2e::common::runner::{setup_package, test_runner};
+
+#[test]
+fn trace_resources() {
+    let temp = setup_package("trace_resources");
+    let snapbox = test_runner();
+    snapbox
+        .current_dir(&temp)
+        .arg("--save-trace-data")
+        .assert()
+        .success();
+
+    for test_name in [
+        "test_call",
+        "test_deploy",
+        // TODO(#1657):
+        //  "test_failed_call",
+        //  "test_failed_lib_call",
+        "test_l1_handler",
+        "test_lib_call",
+    ] {
+        let call_trace = deserialize_call_trace(test_name, &temp);
+        ensure_resources_are_correct(&call_trace);
+    }
+
+    // tests for Deploy, CallContract and LibraryCall syscalls as they cannot be tested as easily as the rest
+    ensure_test_call_syscalls(&temp);
+    ensure_test_deploy_syscalls(&temp);
+    ensure_test_l1_handler_syscalls(&temp);
+    ensure_test_libcall_syscalls(&temp);
```

Comment:
Those should be separate tests. It is hard to understand why we keep all those cases here

Reply:
I can do that